window/src/os/macos/window.rs

```diff
@@ -1496,6 +1496,10 @@ impl WindowView {
         Self::mouse_common(this, nsevent, MouseEventKind::Release(MousePress::Right));
     }
 
+    extern "C" fn other_mouse_up(this: &mut Object, _sel: Sel, nsevent: id) {
+        Self::mouse_common(this, nsevent, MouseEventKind::Release(MousePress::Middle));
```

Comment:
regarding not triggering for non-middle buttons, I think you want something like:

```rust
let button_number = NSEvent::buttonNumber(nsevent);
// if the above doesn't work, do it the hard way: let button_number: isize = msg_send![nsevent, buttonNumber];
if button_number == 1 {
   Self::mouse_common(this, nsevent, MouseEventKind::Release(MousePress::Middle));
}
```

I'm not sure if the middle mouse button is 1 or not, but you can `log::info!("button {}", button_number);` to see what numbers show up.